core/service/src/lib.rs

```diff
@@ -124,7 +124,7 @@ impl<Components> Service<Components>
 	where
 		Components: components::Components,
 		<Components as components::Components>::Executor: std::clone::Clone,
-		for<'de> SignedBlock<<<Components as components::Components>::Factory as components::ServiceFactory>::Block>: ::serde::Deserialize<'de>
+		for<'de> SignedBlock<ComponentBlock<Components>>: ::serde::Deserialize<'de>,
```

Comment:
@bkchr what's the difference between these two? seems pretty subtle...

Reply:
There is no difference and that was not the problem. Look into `core/sr-primitive/src/generic/block.rs`, the signed block deserialize implementation was the problem.

Reply:
ok - so that can be reverted?

Reply:
Can be reverted, but its shorter and doing the same as your code before. So, why revert it?

Reply:
fair enough